src/hgraph/_operators/_tsd_and_mapping.py

```diff
@@ -0,0 +1,77 @@
+from typing import Tuple, Mapping, TypeVar
+
+from hgraph._types import TS, DEFAULT, TSD, K, K_1
+from hgraph._types._time_series_types import TIME_SERIES_TYPE, TIME_SERIES_TYPE_1, OUT
+from hgraph._wiring._decorators import operator
+
+__all__ = ("keys_", "values_", "rekey", "flip", "flip_keys", "partition", "collapse_keys", "uncollapse_keys")
+
+TSD_OR_MAPPING = TypeVar("TSD_OR_MAPPING", TSD, TS[Mapping])
+
+
+@operator
+def keys_(ts: TIME_SERIES_TYPE) -> DEFAULT[OUT]:
+    """
+    Returns the TSS or set of keys in a dictionary.
+    """
+    raise NotImplemented(f"Not implemented for input type {type(ts)}")
+
+
+@operator
+def values_(ts: TIME_SERIES_TYPE) -> DEFAULT[OUT]:
+    """
+    Returns a tuple of the values in the dictionary. Note: does not apply to TSD as there is no suitable time series
+    type
+    """
+    raise NotImplemented(f"Not implemented for input type {type(ts)}")
+
+
+@operator
+def rekey(ts: TIME_SERIES_TYPE, new_keys: TIME_SERIES_TYPE_1) -> DEFAULT[OUT]:
+    """
+    Rekeys the input time series to the provided key.
+    """
+    raise NotImplemented(f"Not implemented for input type {type(ts)}")
+
+
+@operator
+def flip(ts: TIME_SERIES_TYPE) -> DEFAULT[OUT]:
+    """
+    Flips the dictionary so that the values become the keys and the keys become the values.
+    """
+    raise NotImplemented(f"Not implemented for input type {type(ts)}")
+
+
+@operator
+def partition(ts: TIME_SERIES_TYPE, partitions: TIME_SERIES_TYPE_1) -> DEFAULT[OUT]:
```

Comment:
As above (using simplified templates to represented a documented specific signature)